### PR TITLE
[Merged by Bors] - feat: port SetTheory.Game.PGame

### PR DESCRIPTION
**Pull request by maxwell-thum**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/SetTheory/Game/Pgame.lean [hunk 1]
```diff
@@ -0,0 +1,1919 @@
...
+#align pgame.neg_of_lists Pgame.neg_ofLists
+
+theorem isOption_neg {x y : Pgame} : IsOption x (-y) ↔ IsOption (-x) y :=
+  by
+  rw [IsOption_iff, IsOption_iff, or_comm]
+  cases y;
+  apply or_congr <;>
+    · apply exists_congr
+      intro
+      rw [← neg_eq_iff_neg_eq]
```
Comment (MichaelStollBayreuth) [inline Mathlib/SetTheory/Game/Pgame.lean:1249]: `neg_eq_iff_neg_eq` was removed by #2884. Use `rw neg_eq_iff_eq_neg; refl` instead.

File: Mathlib/SetTheory/Game/Pgame.lean [hunk 2]
```diff
@@ -0,0 +1,1919 @@
...
+theorem ofLists_moveRight' {L R : List Pgame} (i : (ofLists L R).RightMoves) :
+    (ofLists L R).moveRight i =
+      R.nthLe (toOfListsRightMoves.symm i) (toOfListsRightMoves.symm i).is_lt :=
+  rfl
+#align pgame.of_lists_move_right' Pgame.ofLists_moveRight'
+
+/-- A variant of `Pgame.recOn` expressed in terms of `Pgame.moveLeft` and `Pgame.moveRight`.
+
+Both this and `Pgame.recOn` describe Conway induction on games. -/
+-- Porting note: Code generator does not support `Pgame.recOn`
```
Comment (vihdzp) [inline Mathlib/SetTheory/Game/Pgame.lean:215]: How does one access the recursion principle of an inductive type in Mathlib 4?

Comment (digama0) [reply Mathlib/SetTheory/Game/Pgame.lean:215]: Same way as in lean 3. Note that the error is that the *code generator* doesn't support the recursor. If you mark it `noncomputable` then it works

**Comment by Ruben-VandeVelde** [comment]
This just needs someone to deal with the termination proofs (which use the custom tactic) at this point, I think

**Pull request closed without merging**